PocketCastsTests/Tests/IAP/IAPHelperTests.swift

```diff
@@ -5,40 +5,30 @@ import StoreKitTest
 import PocketCastsServer
 
 final class IAPHelperTests: XCTestCase {
+    let configurationFile = "Pocket Casts Configuration"
+    let iapTestTimeout: TimeInterval = 5
 
-    var email: String?
+    var session: SKTestSession!
+    var helper: IAPHelper!
 
     override func setUpWithError() throws {
-        // Pretend we're logged in
-        email = ServerSettings.syncingEmail()
-        ServerSettings.setSyncingEmail(email: "test@test.com")
+        session = try SKTestSession(configurationFileNamed: configurationFile)
+        session.clearTransactions()
+        session.resetToDefaultState()
+        session.disableDialogs = true
+
+        helper = IAPHelper(serverHandler: MockIAPHandler())
+        SKPaymentQueue.default().add(helper)
     }
 
     override func tearDownWithError() throws {
-        // Restore previous login
-        ServerSettings.setSyncingEmail(email: email)
+        session.clearTransactions()
+        SKPaymentQueue.default().remove(helper)
     }
 
-    let configurationFile = "Pocket Casts Configuration"
-    let iapTestTimeout: TimeInterval = 5
-
     func testRequestInfo() throws {
-        let session = try SKTestSession(configurationFileNamed: configurationFile)
-        session.clearTransactions()
-        session.resetToDefaultState()
-        session.disableDialogs = true
-
-        let helper = IAPHelper()
-        let expectation = XCTestExpectation(description: "Fetch Products")
-        NotificationCenter.default.addObserver(forName: ServerNotifications.iapProductsUpdated, object: nil, queue: nil) { notification in
-            expectation.fulfill()
-        }
         helper.requestProductInfo()
-
-        wait(for: [expectation], timeout: iapTestTimeout)
-        XCTAssert(helper.hasLoadedProducts)
-
-        let _ = helper.getProduct(for: .monthly)
```

Comment:
I checked the code, but I didn't see a reason for this to be called. Let me know if I missed something.

Reply:
It was just a way to confirm we got the product after the request is completed.